From 79d1a436bb5c7ab40e43130495c455ffc8c682a2 Mon Sep 17 00:00:00 2001
From: Richard Bradfield <bradfirj@fstab.me>
Date: Wed, 22 Jul 2015 01:18:03 +0100
Subject: [PATCH] Replace deprecated tmpnam() call

From Perl 5.22 onwards, POSIX::tmpnam() has been deprecated (without the
usual 2 year deprecation cycle), using the File::Temp module instead
preserves compatibility while allowing compilation on 5.22 and later.
---
 make/utilities.pm | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/make/utilities.pm b/make/utilities.pm
index ae16ce3..baba584 100644
--- a/make/utilities.pm
+++ b/make/utilities.pm
@@ -29,6 +29,7 @@ use warnings FATAL => qw(all);
 
 use Exporter 'import';
 use POSIX;
+use File::Temp;
 use Getopt::Long;
 use Fcntl;
 our @EXPORT = qw(make_rpath pkgconfig_get_include_dirs pkgconfig_get_lib_dirs pkgconfig_check_version translate_functions promptstring);
@@ -404,7 +405,7 @@ sub translate_functions($$)
 			my $tmpfile;
 			do
 			{
-				$tmpfile = tmpnam();
+				$tmpfile = File::Temp::tmpnam();
 			} until sysopen(TF, $tmpfile, O_RDWR|O_CREAT|O_EXCL|O_NOFOLLOW, 0700);
 			print "(Created and executed \e[1;32m$tmpfile\e[0m)\n";
 			print TF $1;
